src/Collector/AbstractCollector.php

```diff
@@ -15,19 +17,31 @@ abstract class AbstractCollector implements CollectorInterface, \Serializable
      */
     protected $data;
 
+    public function __serialize()
+    {
+        return serialize($this->data);
+    }
+
     /**
-     * @see \Serializable
+     * @deprecated since 2.3.0, this method will be removed in version 3.0.0 of this component.
+     *             {@see Serializable} as alternative
      */
     public function serialize()
     {
-        return serialize($this->data);
+        return $this->__serialize();
+    }
+
+    public function __unserialize($data)
+    {
+        $this->data = unserialize($data);
     }
```

Comment:
See above, `$data` should be an array at this point, i.e. no need to call `unserialize()`.